preset.js

```diff
@@ -13,27 +13,28 @@ module.exports = {
 
     existingPlugins.forEach((plugin) => {
       if (!allowedPlugins.includes(plugin) && !options.allowedPlugins.includes(plugin)) {
-        chain.plugins.delete(plugin);
+        chainConfig.plugins.delete(plugin);
       }
     });
 
-    const resolvedConfig = chain.toConfig();
+    const resolvedChain = chainConfig.toConfig();
+    const webpackConfig = api.resolveWebpackConfig();
 
     return {
       ...config,
-      plugins: [...config.plugins, ...resolvedConfig.plugins],
+      plugins: [...config.plugins, ...resolvedChain.plugins],
```

Comment:
I don't like this part. We have to have only one final resolved config. We have two here. If you see [here](https://github.com/vuejs/vue-cli/blob/047872c25eee9f70b5e9e2a67eded2714556e2a7/packages/%40vue/cli-service/lib/Service.js#L229), `resolveWebpackConfig()` does include the chained webpack config. So, my thought is to change the above `resolvedConfig = api.resolveWebpackConfig(resolvedChain)` and use the `resolvedConfig` here. You get what I am saying?

Reply:
I agree we need only one config here. However, the current filter method uses the names from webpack-chain. I don't think we want to filter the config from `resolveWebpackConfig()` since we could only filter by checking if any defined plugin is an instance of a certain plugin class which would mean we could either have no instances of a plugin or all of them. Currently we can filter based on the name we gave the plugin instance with chain-webpack.

The most basic solution would be to only allow plugins defined through chainWebpack.

The most 'complete' option would be to allow both classes and webpack-chain string names to be filtered, but then we'll still need to resolve both configs.

Another options would be to reconsider this feature, what is the reason this filter options was implemented? Is storybook incompatible with standard vue-cli plugins? Otherwise we might create a blacklist or instruct the user not to install certain plugins when building for storybook.

What do you thinks is best?

Reply:
Goo call, pushed.